packages/service-library/src/data-providers/website-provider.ts

```diff
@@ -23,7 +24,23 @@ export class WebsiteProvider {
     public async updateWebsite(website: Partial<Website>): Promise<Website> {
         const websiteDoc = this.normalizeDbDocument(website);
 
-        const response = await this.cosmosContainerClient.mergeOrWriteDocument(websiteDoc);
+        const response = await this.cosmosContainerClient.mergeOrWriteDocument(
+            websiteDoc,
+            websiteDoc.partitionKey,
+            true,
+            (target, source, key) => {
+                if (key === 'knownPages') {
```

Comment:
why we need move this logic out of cosmos client? this should not be a problem to keep it there. at least we can follow lodash pattern and return undef from `if (key === 'knownPages')` function part and keep `if (_.isArray(target))` part in cosmos client. ie if custom merge return undef fallback to cosmos client logic, otherwise use result from custom merge func.